### PR TITLE
Use ALL_NAMESPACES for webhooks-extension state

### LINKED ISSUE #266: Webhooks extension does not show pipeline or service account when namespace on the left is different to the one selected
When creating a webhook you cannot select a pipeline or service account if the namespace you have selected in the create webhook is different to the namespace on the left

### PR DESCRIPTION
**Pull request by ncskier**
# Changes

Fixes #266
The `WebhookApp` was taking its state directly from the Redux store using
the Dashboard's selected namespace (`getSelectedNamespace()`). Instead, we
want to use `ALL_NAMESPACES` so that the `WebhookApp`'s state does not
change when the Dashboard's selected namespace changes.

Also, this PR adds an empty state to the Pipeline and ServiceAccount dropdowns.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: webhooks-extension/src/components/WebhookCreate/WebhookCreate.js [hunk 1]
```diff
@@ -352,7 +352,7 @@ class WebhookCreatePage extends Component {
     return <Dropdown
```
Comment (AlanGreene) [inline webhooks-extension/src/components/WebhookCreate/WebhookCreate.js]: We have a ServiceAccountsDropdown component that already provides an empty state, might be worth switching to that if it's suitable. Alternatively you can use the TooltipDropdown directly which has an `emptyText` prop for this purpose.

File: webhooks-extension/Gopkg.lock [hunk 2]
```diff
@@ -90,6 +90,22 @@
   pruneopts = "UT"
   revision = "6f77996f0c42f7b84e5a2b252227263f93432e9b"
 
+[[projects]]
```
Comment (AlanGreene) [inline webhooks-extension/Gopkg.lock]: please keep dependency updates in separate PRs unless related to the code change, this makes it easier to identify potential issues introduced by the new versions

File: webhooks-extension/src/WebhookApp.js [hunk 3]
```diff
@@ -5,6 +5,8 @@ import { connect } from 'react-redux';
 import { WebhookCreate } from './components/WebhookCreate';
 import { WebhookDisplayTable } from './components/WebhookDisplayTable';
 
+var ALL_NAMESPACES = '*';
```
Comment (AlanGreene) [inline webhooks-extension/src/WebhookApp.js]: use `const` instead of `var`

File: webhooks-extension/src/components/WebhookCreate/WebhookCreate.js (hunk 1 above)
Comment (ncskier) [reply webhooks-extension/src/components/WebhookCreate/WebhookCreate.js]: I actually created issue https://github.com/tektoncd/experimental/issues/304 after working on this one for the same reason as your comment here 😅 

@AlanGreene it was my understanding that the components from the Dashboard weren't currently shared with the Extension, so please let me know if this is incorrect, and if so, how I can use those components.

Comment (AlanGreene) [reply webhooks-extension/src/components/WebhookCreate/WebhookCreate.js]: We publish an npm package with the UI components (https://www.npmjs.com/package/@tektoncd/dashboard-components) - we still need to add documentation and automate the release process but the components should be usable.

However, I forgot that the ServiceAccountsDropdown isn't included as it's a container (connected to the store) 😢 It would be relatively simple to move the TooltipDropdown to the components package though.

Comment (ncskier) [reply webhooks-extension/src/components/WebhookCreate/WebhookCreate.js]: Ok, I'll hold off on updating the components in this PR, and leave it to issue #304 if that's alright with you @AlanGreene.

**Comment by ncskier** [comment]
Thanks for the feedback @AlanGreene. I think I've addressed all of your comments, but let me know if there's anything else I can do 🙂

File: webhooks-extension/src/components/WebhookCreate/WebhookCreate.js (hunk 1 above)
Comment (ncskier) [reply webhooks-extension/src/components/WebhookCreate/WebhookCreate.js]: Is it possible to share the Containers since they're connected to the store, or only the components?

Comment (AlanGreene) [reply webhooks-extension/src/components/WebhookCreate/WebhookCreate.js]: It would be possible although a little awkward, and likely of limited value. The containers make assumptions that may not be valid for 3rd party consumers outside the dashboard application and actually using these published containers would probably require similar amount of code as writing your own using the existing UI components.
